### DISCUSSION TITLE
ImageMagick command-line

### DISCUSSION THREAD

**Discussion by gsmith0717** [ImageMagick/ImageMagick#2406] [Help]
Hi all,
According to the documentation, the ImageMagick command-line consists of

1. one or more required input filenames.
2. zero, one, or more image settings.
3. zero, one, or more image operators.
4. zero, one, or more image sequence operators.
5. zero, one, or more image stacks.
6. zero or one output image filenames (required by magick, convert, composite, montage, compare, import, conjure).

My question is regarding #6 the output fille name.    Can anyone provide examples or provide documentation that shows the  ImageMagicK command-line that uses Zero output image filenames.  All the commands that I try require an output image filename.   

thanks,
JG

**Comment by fmw42** [comment]
Processing an image to get statistics or do computations does not write an output. Also you can use null: for the output to output no images. For statistics and such see string formats or escapes at https://imagemagick.org/script/escape.php. For computations see fx: at https://imagemagick.org/script/fx.php and https://imagemagick.org/Usage/transform/#fx_escapes

example to get the image width and height

```
convert rose: -format "%wx%h" info:
70x46

or 

identify -format "%wx%h" rose:
70x46

```


And mogrify does not have an output image

`mogrify -format png *.jpg
`

**Comment by GeeMack** [comment]
Off the top of my head I can think of a couple ways to use command line IM without an output file name. As fmw42 described, a command that does some extensive analysis and outputs the result to the console might look like this...

    convert input.png -verbose info:

It's not technically an output file, but the "info:" can be thought of as an output name.

The output from IM can also be fed into a pipe. I often run single IM commands that create multiple images and pipe them into a command line program, "ffmpeg" in this example, to produce a video.

    convert -size 200x200 xc:red xc:blue xc:yellow -duplicate 100,0,1,2 png:- | ffmpeg -i - output.mpg

The "png:-" there specifies the output should be in the PNG format and fed to a pipe. So that uses no actual ImageMagick output file or filename, too.

**Comment by snibgo** [comment]
I would say that "magick" and "convert" normally need an output filename, but this need not be an _image_ filename. As Fred says, this can be "null:" or "info:".  As GeeMack says, this can be "-" which is a type of filename, sort of.

Strictly speaking, v7 "magick" doesn't need to end with an output filename. Instead, it can end with "-exit".
```
magick xc:White -exit
```
This is a valid command, though somewhat pointless. A command ending with "-exit" would normally "write" some output before that.

**Reply by GeeMack** [reply]
I've used IM to do an occasional calculation at the command line just because I already know part of an FX formula.

    convert xc: -print %[fx:max(432/pi,100)] null:
